jumpscale/clients/sendgrid/sendgrid.py

```diff
@@ -0,0 +1,52 @@
+import base64
+import os
+import io
+import sendgrid
+from sendgrid.helpers.mail import Mail, Attachment
+from python_http_client.exceptions import HTTPError
+from jumpscale.god import j
+from jumpscale.clients.base import Client
+from jumpscale.core.base import fields
+
+
+class SendGridClient(Client):
+    apikey = fields.String()
+
+    def __init__(self):
+        super().__init__()
+
+    def build_attachment(self, filepath, typ="application/pdf"):
+        """
+        Returns a valid sendgrid attachment from typical attachment object.
+        """
+        data = io.BytesIO()
+        with open(filepath, "rb") as f:
+            while True:
+                d = f.read(2 ** 20)
+                if not d:
+                    break
+                data.write(d)
+        data.seek(0)
+        attachment = sendgrid.Attachment()
+        # attachment.set_content(base64.b64encode(data))
+        attachment.file_content = j.data.serializers.base64.encode(data.read()).decode()
+        attachment.file_type = typ
+        attachment.file_name = os.path.basename(filepath)
+        attachment.disposition = "attachment"
+        return attachment
+
+    def send(self, sender, subject, html_content="<strong>Email</strong>", recipients=[], attachments=[]):
```

Comment:
Even if your are sure it's safe to use empty list as a default argument, we don't need this.